### ISSUE TITLE
Email Verification: Improve the UI of the Pending Email Change banner

### ISSUE THREAD

**Issue by javierarce** [Automattic/wp-calypso#94247]
Related to #93895

When a user with an unverified email address changes it, we currently display two notifications stacked on top of each other, which creates a confusing experience for the user.

![image](https://github.com/user-attachments/assets/f5077fc1-f1f6-4bf7-8632-b63d27d3f820)

We are addressing the placement of the notification on top in #93895. This issue focuses on the one at the bottom, the pending email change banner.

The proposal is to replace the second notification and use the line under the email field to offer users the option to update their address.

![image](https://github.com/user-attachments/assets/158eb6e9-efe2-40be-bc7a-bcdc2aa2d79c)

Clicking on the link will enable the input field so they can modify the email:

![image](https://github.com/user-attachments/assets/c18fd35b-84d4-47c7-855c-5f586a34ddad)

Once the user submits the form, we display a floating notification to inform them that we have sent a verification request to their email account:

![image](https://github.com/user-attachments/assets/0440249b-e322-4d29-8c2f-bafbebe03119)

This way, we'll also make cancel functionality we offer in the curent banner more transparent for the user.

Link to the designs: wCm5Tzb7JBaG4WNowWbjES-fi-4978_968

**Comment by Addison-Stavlo** [comment]
@javierarce - This second notice has another variation with some extra format and copy in the case of having a custom domain registration:

<img width="954" alt="Screenshot 2024-09-18 at 2 03 32 PM" src="https://github.com/user-attachments/assets/d10e8ec2-be60-4f00-99fa-486b6567b5ab">

Should we add an extra line in the copy under the input in this case? Or other thoughts?

**Comment by Addison-Stavlo** [comment]
Another clarification Q in addition to the above @javierarce : The banner component currently displays interface color (--color-accent). It seems like we are kind of using it as a warning notice instead. Should we override it to be red like in your design (or is that actually your accent color?) ?

<img width="811" alt="Screenshot 2024-09-18 at 4 18 28 PM" src="https://github.com/user-attachments/assets/9093583a-8a56-4e52-8039-737e41df4b68">

Im assuming we may want it to be red given this context regardless of interface colors, but figured I would confirm.

Q: Also note in the copy there is a grammatical error around "your email help you secure". Should this be "your email helps you secure" or "your email will help you secure" ?

**Comment by Addison-Stavlo** [comment]
@javierarce for when you are back. I have 2 separate items to consider for you.  

### 1. Explination copies
Related to [my comment two comments above this](https://github.com/Automattic/wp-calypso/issues/94247#issuecomment-2359103290), I tried to adjust the explanation copy to accommodate these extra cases within the pending email change cases:
1. If the user has a custom domain, there is extra information given about needing to update the email for those registrations as well.
2. It wasn't clear (to me) by the original notice, but the "cancel" button actually is set up to cancel the pending email change (go back to the verified email), and unlock the input. I added a a section of the explanation to show a button to cancel the pending email change in the case that there is an email change pending. I also think this could allow us to simplify a bit (more details about this in item 2 below).

Ive done my best to fit these extra cases and contexts into the explanation text shown. These are how I have currently setup those cases in the [PR](https://github.com/Automattic/wp-calypso/pull/94683) ( I imagine we might want to adjust, consolidate, remove - but I wanted to first highlight the functionality that is in place before ripping it out):

Pending email change (no domains):
<img width="653" alt="Screenshot 2024-09-20 at 2 55 00 PM" src="https://github.com/user-attachments/assets/ec54c6db-eb57-4714-8353-545be814aa41">

Pending email change (with domains):
<img width="685" alt="Screenshot 2024-09-20 at 2 56 45 PM" src="https://github.com/user-attachments/assets/0905f29c-d325-4c33-9fc1-730b8f9621ab">

Email not verified:
<img width="691" alt="Screenshot 2024-09-20 at 2 57 36 PM" src="https://github.com/user-attachments/assets/987f37bc-9776-428b-ad3c-3c92c823ae1b">

General case:
<img width="697" alt="Screenshot 2024-09-20 at 2 57 53 PM" src="https://github.com/user-attachments/assets/393ea2a4-4971-4d1d-8c52-a734f519bd1d">

### 2 Simplification - don't allow unlocking the input in this case.

After looking into all this there is one proposal I would like to make, as this seems like it is starting to feel more complicated than necessary on the code side (and maybe the UX side?) in order to enable unlocking and editing during pending changes, etc.

My proposal is that we continue to use the **existing behavior** for when the input is unlocked vs. locked.
* No pending email change: input is enabled.
* Pending email change: input is disabled.

So instead of us having a link to "click here to update it" to enable the input, we instead lean on a link to "click here to cancel the pending email change" (which in turn does revert the pending change and unlocks the input). It seems subtle, but I think this would simplify the behavior a lot, as well as the code and the copies shown.

New copies could then be along the lines of:

* Pending email change (no domains):
"Your new email has not been verified yet. To cancel the pending email change <u>click here</u>.
Update contact information on your domain names if necessary <u>here</u>."

* Pending email change (with domains):
"Your new email has not been verified yet. To cancel the pending email change <u>click here</u>."

* Email not verified:
"Your email has not been verified yet."

* General case:
"Not publicly displayed, except to owners of sites you subscribe to."

Thoughts? [This PR](https://github.com/Automattic/wp-calypso/pull/94683) is currently in the unsimplified form if you would like to test the calypso live link, but Im starting to think we may want to simplify that lock/unlock behavior back to whether or not there is a pending change.

**Comment by Addison-Stavlo** [comment]
### Re - cancel pending vs. unlock and submit new

> do we really need to give users the option to cancel the pending email change? 

Generally speaking, the user should be able to cancel a pending email change here if they no longer want it. 

But even if we weren't enabling that I think clarity can be good here. There isn't much in this new design differentiating original verification from pending email change, and when a change is pending we don't do much to communicate it (even though the email we show in the input is an unverified email that is NOT yet set as their account email). So I think leaning towards embracing this 'cancel pending' could do a lot for keeping necessary functionality, adding clarity, and simplicity in the input behavior.

> My proposal was to disable the email field, 

The field is already disabled when there is a pending change. 👍 

> and if they click on "click here to change it" and submit a new email, we would cancel the previous request automatically.

Currently cancelling the pending change is what we allow enabling the field based on. Can we continue to do that? Or is there a need for the user to actually unlock the field and submit another address before we cancel the currently pending email change? I think the former is more simple, but technically we could support the latter. But in general we should support cancelling a pending change without requiring a new pending change. 

Recapping all of the above from the code and ux side I think id strongly prefer the "cancel pending change" to unlock the field functionality.

### Re updating domains info

> I think displaying that message before the user confirms their email address is a bit confusing: we are asking them to do two things in different places, and the second action only makes sense if they can actually confirm their email address.

That is very true.

>  I think a better approach could be to display the message in the snackbar message they see after they have confirmed their email account:

That could be! Although I may want to avoid that in this current PR to avoid inflating the scope even more. It seems like something we could continue with here though.

**Comment by Addison-Stavlo** [comment]
@javierarce another question separate to the above. How do you think we should handle the "click here to update it" link on the dialog/modal when the user is already on the account page with the email form?

![image](https://github.com/user-attachments/assets/1433e75b-5759-4df8-8fd3-e096f93e6427)

Do we hide that part of the message when we are on the page already? Or something else?

**Comment by javierarce** [comment]
Regarding the "cancel pending vs. unlock and submit new" part of your comment: I'm a bit confused about the options we have and the importance of the cancelling functionality, so please let me explain my current understanding of the whole flow again and also my proposal:

- If a user creates an account without verifying their email address, we display a message asking them to check their inbox to verify their email.
- Users can do two things:
	1. Request a new verification email to be sent to the address they originally used to register their account. 
	2. Update the email address (this is useful if, for example, they made a spelling mistake or can't access the original email account).
- In the current UI, if they modify the email address, we display a message that says: "Your email change is pending. Please check (email address) for an email with the subject 'WordPress.com: New Email Address' to confirm the change."
- The users then have two options:
	1. Go to the updated email address inbox and check the email.
	2. Cancel the request.
- If they open the email and click on the main link, they'll verify the new email address.
- If they cancel the request, we'll enable the email field and populate it with their previous email address.

What I was proposing is to hide this complexity from the user in the following way:

- If the email is not verified, we disable the field and display a message saying "Your email has not been verified yet. Need to update your address? Click here to change it"
- If the user clicks that link and changes the email address, we send them the verification email to that address.
- If they realized the made a mistake, they can click on the link again, edit their email address and once they submit the form we'll send them a new email to the new address.

There's another extra simplification here: 

- We could enable the email field and just display this message below: "Your email has not been verified yet." and let the user modify it freely whenver they want. 

Could any of these two options work? Do you think it could impact user experience negatively?

**Comment by javierarce** [comment]
> Do we hide that part of the message when we are on the page already? Or something else? It doesn't exist on that page yet so there isn't really a current behavior to compare, but it will exist there in the new PR since we are adding it as part of the banner.

Could we scroll down to the email field when they click on that link? I think this could be the common behavior for that link in these modals, regardless of where they appear.

**Comment by Addison-Stavlo** [comment]
> my current understanding of the whole flow again

This is my understanding of the current behavior as well.

> What I was proposing is to hide this complexity from the user in the following way:

I think its important to highlight distinct differences between email verification and email pending change:
* Email verification - Is the initial verification after account creation. The email shown is the accounts `user_email` field, but there has been no verification at all.
* Email pending change - The email shown is NOT the accounts `user_email` field, and is simply the email that the user prompted to change their email to but never confirmed. The account email has not actually been changed yet. This is different than verification because 1 - its not verifying the existing account email, but completing the process of changing the account email, 2 - verification of the new email comes intrinsically with completion since the completion is done through that new email. So this is actually not 'verifying their email' at all, its 'confirming to to have the accounts email address be updated to a new value'.

> Do you think it could impact user experience negatively?

Yes. We would be showing an email address in a locked field that is NOT their current account email address and we will no longer be showing any indication of that. This seems incorrect and misleading.

> If they realized the made a mistake, they can click on the link again, edit their email address and once they submit the form we'll send them a new email to the new address.

One sticky point in this is going back to their current account email address. So say a user:
1. Changes the email address to a new address in the form. The email is now pending change, the form is locked and showing the pending email instead of the account email.
2. Later, the user **_now wants to cancel the pending change and continue using their account email on file_**. The user then has to click the "change it here" button you propose.
3. Now instead of simply cancelling the pending the change, the user needs to type their old address in specifically in the field and submit it, instead of just hitting a button to cancel the pending change. 
    * We could also imagine an edge case here where a user may not remember which of their multiple emails was the one currently on file for their account to type it in manually.
4. Say this worked and the user typed in their current account email and submitted the form to undo the pending change. The email in the input changed due to this. However, there will not be a new email for them to confirm so we don't prompt them with an email address to check. The account email wasn't changed, so the success notices for changed/updated settings aren't relevant. But the email shown in the input changed, so it seems like we should say something, which would then be about cancelling the pending change and surfacing that level complexity anyways 😅.

3 gets a little sticky on the code side as well. We are telling the system to change the account email from what it currently is, to exactly what it currently is (remember the other change was still pending, its not yet the account email!). There is a bit of structure in place to prevent that redundancy and clears the unsaved changes if the suggested change is equal to the current value. So we would have to add a bit of ad-hoc handling to hack around this general settings functionality to allow a user to cancel a pending email change by manually typing in their current on-file email, and now have an ad-hoc setting field on the code side that has weird handling over if its value is equal to its current value and needing to mutate/clear a separate setting (the pending email address) in some cases... it seems like something to avoid unless there is a strong reason to support it. As opposed to keeping it clean and unconflated as it currently is: cancelling the pending change clears the pending change, the email setting behaves the same as the rest of the settings in the api, etc. UX concerns aside this seems much more preferred and maintainable on the code side.

I like the idea to reduce complexity in front of the user, but I think there are some distinct differences between verification and email change and that trying to hide those will result in more complexity in the end. Having the cancel functionality unlock the form keeps a lot of complexity down and negates a lot of edge cases (and potential user facing bugs) that present themselves once we start getting into updating the field again while a pending change is active.

**Comment by Addison-Stavlo** [comment]
> Could we scroll down to the email field when they click on that link? I think this could be the common behavior for that link in these modals, regardless of where they appear.

We certainly could, but note that for me right now that field is near the top of the page so its already in scroll view. It kind of just feels like clicking it dismisses the modal without doing anything, which may be ok? 🤷‍♀️

**Comment by javierarce** [comment]
Thanks for that response, @Addison-Stavlo, I'm convinced by your explanation. Let's proceed with your simplified approach. I've updated the designs here wCm5Tzb7JBaG4WNowWbjES-fi-5057_13109, but the made the cancellation prompt a bit more straightforward: "Your email has not been verified yet. Cancel the pending email change". 

![image](https://github.com/user-attachments/assets/678d02de-3766-4fca-a11e-2c1ce8eb7720)

And regarding the scrolling behavior, I did a quick test and I think it could work. Focusing on the email might make it more evident what has happened. What do you think?

https://github.com/user-attachments/assets/5a143ea5-3664-436e-9868-879830e1ba0d

**Comment by Addison-Stavlo** [comment]
Sounds good! Thanks for your patience working through this 😄 

> And regarding the scrolling behavior, I did a quick test and I think it could work.

Yeah we can give that a try and see how it feels.

> Focusing on the email might make it more evident what has happened. What do you think?

This definitely makes it visible, but the email is disabled when pending change so not focusable in those circumstances. Another option could be to focus the 'cancel the pending change' button in that case, what do you think? I don't think it has a normal focus style that shows up in non-a11y flows but maybe we could enable one specifically when this action happens.

**Comment by Addison-Stavlo** [comment]
https://github.com/Automattic/wp-calypso/pull/94683 should now have all the changes we have discussed.

**Comment by javierarce** [comment]
> Sounds good! Thanks for your patience working through this 😄

I should say the same!

> This definitely makes it visible, but the email is disabled when pending change so not focusable in those circumstances. Another option could be to focus the 'cancel the pending change' button in that case, what do you think? I don't think it has a normal focus style that shows up in non-a11y flows but maybe we could enable one specifically when this action happens.

I just tried https://github.com/Automattic/wp-calypso/pull/94683 and I think your solution  works perfectly well in both cases, so I think we could keep it that way.
